components/esmterrain/storage.cpp

```diff
@@ -353,20 +355,18 @@ namespace ESMTerrain
     std::string Storage::getTextureName(UniqueTextureId id)
     {
         // Goes under used terrain blend transitions
-        static const std::string baseTexture = "textures\\tx_black_01.dds";
         if (id.first == -1)
-            return baseTexture;
+            return mBaseTexture;
```

Comment:
Considering `baseTexture` is `static const`, there was no problem here or below with the same construct. Static variables are guaranteed to be initialized in a thread-safe manner, and as the variable's `const` with no mutable members, it can't be modified. It's only read from to make a copy, so there's no race (multiple threads can read the same memory at the same time unsynchronized just fine).

Reply:
> Considering baseTexture is static const, there was no problem here or below with the same construct. 

GCC does not think so.

Reply:
Interesting - it does not compain if I initialize variable during compilation:

```
static constexpr const char* baseTexture = "textures\\tx_black_01.dds";
```

Reply:
Probably because that causes the returned `std::string` to be constructed in the given thread from a const char array, rather than copy-constructing in the given thread from a separate `std::string` object. A static const/constexpr string can also be initialized at load time since it doesn't depend on runtime state, unlike a string which needs to be initialized when it's first executed since it does depend on runtime state (the allocator).

Given that the function returns a new string object, rather than a const reference, constructing it from a char array shouldn't be any worse (might even be better) than copy-constructing it from another string. In such a case, I'd recommend:
```
static constexpr char baseTexture[] = "textures\\tx_black_01.dds";
```
And similar for the other static const string.